### PR TITLE
fix test suite warnings re `drop`

### PR DESCRIPTION
**Pull request by max-sixty**
Some more warnings silenced

### CODE REVIEW

**Comment by max-sixty** [comment]
ref https://github.com/pydata/xarray/issues/3266. This was causing about a quarter of the total, because of the test's parameterization.

File: xarray/tests/test_duck_array_ops.py [hunk 1]
```diff
@@ -440,7 +440,9 @@ def test_argmin_max(dim_num, dtype, contains_nan, dask, func, skipna, aggdim):
             **{aggdim: getattr(da, "arg" + func)(dim=aggdim, skipna=skipna).compute()}
         )
         expected = getattr(da, func)(dim=aggdim, skipna=skipna)
-        assert_allclose(actual.drop(actual.coords), expected.drop(expected.coords))
+        assert_allclose(
+            actual.drop(list(actual.coords)), expected.drop(list(expected.coords))
```
Comment (dcherian) [inline xarray/tests/test_duck_array_ops.py]: do we want to special case the passing of coordinate objects for  convenience?

Comment (max-sixty) [reply xarray/tests/test_duck_array_ops.py]: It's a classic special case question! https://github.com/pydata/xarray/pull/3128/files#r315000922

Comment (max-sixty) [reply xarray/tests/test_duck_array_ops.py]: I hear you, but I can also see: https://github.com/pydata/xarray/pull/3128/files#r315000922

Comment (max-sixty) [reply xarray/tests/test_duck_array_ops.py]: Fair point, but I can also see this POV: https://github.com/pydata/xarray/pull/3128/files#r315000922 ; I defer to @shoyer

**Comment by max-sixty** [comment]
One other issue: it suggests using `drop_vars` but that doesn't exist!

**Comment by shoyer** [comment]
> One other issue: it suggests using `drop_vars` but that doesn't exist!

Yes, this was a suggestion for a  new method

File: xarray/tests/test_duck_array_ops.py (hunk 1 above)
Comment (dcherian) [reply xarray/tests/test_duck_array_ops.py]: Ah wasn't aware. That makes sense.

**Comment by max-sixty** [comment]
> Yes, this was a suggestion for a new method

How would this work differently from `.drop(var_name)`? Or would we deprecate that use of `drop`?

**Comment by shoyer** [comment]
I think the idea is just to provide a more explicit option for dropping
variables instead of labels. It's a little messy all in one API.

**Comment by max-sixty** [comment]
> I think the idea is just to provide a more explicit option for dropping variables instead of labels. It's a little messy all in one API.

Yeah, agree it's messy. I think we could have `drop_vars`, completely deprecate any use of `drop` apart from lables, and then we're in a decent place? Or is that too much churn?

**Comment by shoyer** [comment]
Yeah, that was my thought

**Pull request merged**